src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputConstantConverter.cs

```diff
@@ -109,10 +109,8 @@ public static object ReadConstantValue(ref Utf8JsonReader reader, string propert
                 case InputLiteralType literalType:
                     value = literalType.Value;
                     break;
-                case InputModelType model:
-                    throw new JsonException($"Not supported type: InputModelType");
                 default:
-                    throw new JsonException($"Not supported type: {type.Name}");
+                    throw new JsonException($"Not supported type: {type.GetType()}");
```

Comment:
What's the output of `type.GetType()`? I didn't see an overwrite of `GetType()` in `InputType`.

Reply:
when you call `obj.GetType`, it always returns its type (this method is defined on `object`), it is polymorphism in OOP